0n0-CART.Rmd

```diff
@@ -0,0 +1,263 @@
+# Cross validation
+cvTree = cv.tree(classTree, method = 'misclass')
+cvTree
+plot(prune$size, prune$dev, main = "Scree/Elbow Plot", xlab = "Number of Leaves", ylab = "Misclassifications", type = 'b')
+points(cvTree$size, cvTree$dev, type = 'b', pch = 2)
+legend("topright", pch = 1:2, legend = c("Pruned", "Cross Validated"), bty = 'n')
+```
+
+Other methods use a significance test approach to determining whether or not to split. For each variable, the model performs a univariate hypothesis test and splits on the variable with the lowest p-value. If no null hypotheses can be rejected, the tree stops splitting.
+
+```{r}
+library(party)
+ciTree = ctree(Species ~ Sepal.Length + Sepal.Width + Petal.Length + Petal.Width, data = iris)
+ciTree
+plot(ciTree)
+
+```
+
+A main advantage of trees is their ease of interpretability and use. 
+```{r}
+install.packages("DAAG")
+library(DAAG)
+data(spam7)
+spamTree = ctree(yesno ~., data = spam7)
+spamTree
+plot(spamTree)
+```
+How well does it predict?
+```{r}
+spamTreeConfusionMatrix = table(predict(spamTree, spam7), spam7$yesno)
+spamTreeConfusionMatrix
+sum(diag(spamTreeConfusionMatrix))/sum(spamTreeConfusionMatrix) #86.7%
+
+# Let's do a simple cross validation
+
+inSampleProp = .85
+inSampleIndicator = sample(c(TRUE, FALSE), size = nrow(spam7), replace = TRUE, prob = c(inSampleProp, 1 - inSampleProp))
+trainingSet = spam7[inSampleIndicator,]
+testingSet = spam7[!inSampleIndicator,]
+trainingTree = ctree(yesno ~., data = trainingSet)
+trainingTree
+plot(trainingTree)
+
+```
+Now we can do a more fair out of sample calculation
+```{r}
+# In-sample
+inSampleMat = table(predict(trainingTree, trainingSet), trainingSet$yesno)
+inSampleMat
+sum(diag(inSampleMat))/sum(inSampleMat)
+
+# Out of sample
+outSampleMat = table(predict(trainingTree, testingSet), testingSet$yesno)
+outSampleMat
+sum(diag(outSampleMat))/sum(outSampleMat)
+
+```
+
+
+# Bootstrapping
+
+How can we estimate the variability in a statistic? Let's say we want a probability interval for the mean of some data.
+```{r}
+n = 100
+simData = rgamma(n, 2, 1)
+# True mean: 2
+hist(simData)
+mean(simData)
+```
+One option is to rely on normal theory and write,
+
+$$var(\bar{x}) = \frac{1}{n}var(x),$$
+then use the right normal score to get the ``confidence interval'' you want:
+
+```{r}
+xbar = mean(simData)
+xsd = sd(simData)
+# 95% interval
+confInt = xbar + c(-1.96, 1.96) * (xsd/sqrt(n))
+confInt
+```
+The accuracy of this estimate depends on the extent to which the data are normal. We know the data are from a rather skewed gamma, how much does that skewness affect the interval? How could we find out?
+
+What if we had some large number of samples, say m,  of the mean computed from $n$ samples of this population? This is an awful idea in practice, but bear with me.
+```{r}
+m = 2000
+simulatedMeans = numeric(m)
+for(i in 1:m) simulatedMeans[i] = mean(rgamma(n, 2, 1))
```

Comment:
you can write this in vectorized form :p

Reply:
I had it as an sapply but didn't want to have more syntax than needed, but thanks <3.